File: cppFinal/IceCream.cpp
#include "IceCream.h"
#include "iostream"
using namespace std;

IceCream::IceCream(string flavor_)
    : flavor(flavor_) {

}

void IceCream::addTopping(string topping) {
    this->topping.addTopping(topping);
}

void IceCream::display() {
    cout << "The flavor of this ice cream is " << flavor << endl;
    this->topping.display();
}

IceCream::~IceCream() {
    
}

File: javaFinal/src/javaFinal/IceCream.java
package javaFinal;

public class IceCream implements iIceCream {

	private String flavor;
	private Topping topping;
	
	public IceCream(String flavor) {
		this.flavor = flavor;
		this.topping = new Topping();
	}
	
	public void addTopping(String topping) {
		this.topping.addTopping(topping);
	}
	
	public void display() {
		System.out.println("The flavor of this ice cream is " + flavor + ".");
		topping.display();
	}

}
